### PR TITLE
[SDP-1413] add e2e test for PHONE_NUMBER_AND_WALLET_ADDRESS

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Update integration tests to include the test case where we use `registrationContactType=PHONE_NUMBER_AND_WALLET_ADDRESS`. This includes updating the GH Action and the e2e shell script.

Also, reorganized the integration tests code a bit.

### Why

Address https://stellarorg.atlassian.net/browse/SDP-1413

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [x] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: cmd/integration_tests.go [hunk 1]
```diff
@@ -114,7 +114,6 @@ func (c *IntegrationTestsCommand) Command() *cobra.Command {
 			CustomSetValue: cmdUtils.SetRegistrationContactType,
 			ConfigKey:      &integrationTestsOpts.RegistrationContactType,
 			Required:       true,
-			FlagDefault:    data.RegistrationContactTypePhone.String(),
```
Comment (marcelosalloum) [inline cmd/integration_tests.go:117]: I removed this default value to enforce users to explicitly pass the value they want.

File: internal/integrationtests/scripts/e2e_integration_test.sh
Comment (marcelosalloum) [inline internal/integrationtests/scripts/e2e_integration_test.sh:1]: In this file I removed the default value (`export REGISTRATION_CONTACT_TYPE="PHONE_NUMBER"`) and used an object-like approach with the `options`.

**Review by marcelosalloum — COMMENTED** [review]
I'm highlighting some of the fixes in the comments below.

**Review by marwen-abid — APPROVED** [review]
LGTM! Great refactor

**Pull request merged**